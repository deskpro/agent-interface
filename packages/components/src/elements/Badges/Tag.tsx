import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/dp-badges.css";

export type TagProps = {
  color?: DPColor;
  size?: "small" | "medium";
};

const Tag: React.SFC<TagProps> = ({ children, color, size }) => (
  <span
    className={classNames("dp-Tag", {
      [`Tag--${color}`]: !!color,
      [`Tag--${size}`]: !!size
    })}
  >
    {children}
  </span>
);

export default Tag;
