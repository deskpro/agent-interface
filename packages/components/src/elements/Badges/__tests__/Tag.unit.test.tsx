import * as React from "react";
import { render } from "react-testing-library";

import Tag from "../Tag";

describe("<Tag />", () => {
  it("should match its snapshot", () => {
    const { getByText } = render(<Tag>Tag text</Tag>);
    expect(getByText("Tag text")).toMatchSnapshot();
  });
});
