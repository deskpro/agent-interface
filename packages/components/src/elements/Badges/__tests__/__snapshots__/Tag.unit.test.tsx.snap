// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> it should match its snapshot 1`] = `
<span
  class="dp-Tag"
>
  Tag text
</span>
`;
